kazel/generator.go

```diff
@@ -56,7 +56,7 @@ func (v *Vendorer) walkGenerated() error {
 // returns the go import paths. It does not follow symlinks.
 func (v *Vendorer) findOpenAPI(root string) ([]string, error) {
 	for _, r := range v.skippedPaths {
-		if r.MatchString(root) {
+		if r.MatchString(filepath.Base(root)) {
```

Comment:
This is inconsistent with the other usages:
https://github.com/kubernetes/repo-infra/blob/d9bb9fdc907665c61c228baa64fed9b91c7dc1b0/kazel/kazel.go#L230-L233
https://github.com/kubernetes/repo-infra/blob/d9bb9fdc907665c61c228baa64fed9b91c7dc1b0/kazel/sourcerer.go#L43-L47

`SkippedPaths` is intended to be a regex that matches the full path; we should probably document that better, and possibly also write a utility function, rather than reproducing this logic in 3 different places.

If we want to skip any directory with an `_`, it'd be better to update the `SkippedPaths` regexp in kubernetes/kubernetes to not anchor to the start of the path.

Reply:
Alright. In that case, these need to be updated as well - https://github.com/kubernetes/repo-infra/blob/2a736b4fba317cf3038e3cbd06899b544b875fae/kazel/kazel.go#L134

Reply:
no, those are correct - the `.git` and `bazel-*` directories only appear at the workspace root.